rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,532 @@
+                "your training data. To extract non-pretrained entities, add one of "
+                f"{types_to_str(TRAINABLE_EXTRACTORS)} to your pipeline."
+            )
+
+        if training_data.entity_examples and self._component_types.isdisjoint(
+            {DIETClassifierGraphComponent, CRFEntityExtractorGraphComponent}
+        ):
+            if training_data.entity_roles_groups_used():
+                rasa.shared.utils.io.raise_warning(
+                    f"You have defined training data with entities that "
+                    f"have roles/groups, but your NLU pipeline does not "
+                    f"include a '{DIETClassifierGraphComponent.__name__}' "
+                    f"or a '{CRFEntityExtractorGraphComponent.__name__}'. "
+                    f"To train entities that have roles/groups, "
+                    f"add either '{DIETClassifierGraphComponent.__name__}' "
+                    f"or '{CRFEntityExtractorGraphComponent.__name__}' to your "
+                    f"pipeline."
+                )
+
+        if training_data.regex_features and self._component_types.isdisjoint(
+            [RegexFeaturizerGraphComponent, RegexEntityExtractorGraphComponent],
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"You have defined training data with regexes, but "
+                f"your NLU pipeline does not include a 'RegexFeaturizer' or a "
+                f"'RegexEntityExtractor'. To use regexes, include either a "
+                f"'{RegexFeaturizerGraphComponent.__name__}' or a "
+                f"'{RegexEntityExtractorGraphComponent.__name__}' in your pipeline."
+            )
+
+        # NOTE: bugs
+        # 1. removed the check that complains if neither "CRFEntityExtractor",
+        #   nor "DIETClassifier" are contained in case a lookup table is given...
+        #   ... because
+        #   - CRFEntityExtractor: only uses lookup tables if a RegexFeaturizer is
+        #       present *and* the patterns feature is defined (see also 2.)
+        #   - DIET: does not use the lookup table...
+        # 2. previously we warned if there is a lookup table and the
+        #    CRF entity extractor has no "patterns" defined
+        #    a) this is optional but not mandatory, so we *could* remove that warning
+        #       (it is still below - see if "CRFEntityExtractor...")
+        #    b) we definitely should warn if "patterns" is defined but there
+        #       is no regex featurizer
+        #       (this is a new warning added under the if "CRFEntityExtractor...")
+
+        if training_data.lookup_tables and self._component_types.isdisjoint(
+            [RegexFeaturizerGraphComponent, RegexEntityExtractorGraphComponent],
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"You have defined training data consisting of lookup tables, but "
+                f"your NLU pipeline does not include a "
```

Comment:
maybe let's not use pipeline as this would outdate as the configuration format changes, what do you think? 
```suggestion
                f"your NLU configuration does not include a "
```

Reply:
Wasn't sure what to use. In some places, I called it "graph". Will  (try to) switch to "configuration" (everywhere). Seems the most generic.

Reply:
I personally find it very hard to deal with old vs new names. E.g. "NLU components" just perfectly describes which part in the graph I mean but it technically doesn't make a lot of sense anymore 🙈

Reply:
I think if we use "NLU components" for just the components and "NLU configuration" for the part of the graph that is applied per message (and "Core's NLU configuration" for the part of the "Core configuration using NLU components", then we'd have an unambiguous naming convention 🤔  - Not saying it's a good one :D